### PR TITLE
feat: support `wrapWidth` option

### LINKED ISSUE #106: Idea: Wrap width
It would be nice to be able to specify a line width (in number of characters) for the formatter to automatically break long lines on word boundaries before the width is overreached. This should _not_ be applied to `pre` formatted blocks, of course.

* The primary motivation is generating output for plain text editors.
* It is possible to re-parse `mdream`'s output and apply line wrapping using another library, but that would mean another (costly) parsing / processing step.
* There probably isn't a "one true way" to do the wrapping, so some trade-offs will have to be taken (`html-to-markdown` does a decent job, though.); ideally the algorithm can be tweaked.
* no additional runtime cost should be paid if no wrap width is requested

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

Closes #106

### ❓ Type of change

- [ ] 📖 Documentation
- [ ] 🐞 Bug fix
- [ ] 👌 Enhancement
- [x] ✨ New feature
- [ ] 🧹 Chore
- [ ] ⚠️ Breaking change

### 📚 Description

Adds a `wrapWidth` option that hard-wraps prose at a character width on word boundaries, for generating output aimed at plain-text editors (#106). It runs inline during conversion rather than as a second pass, so it costs nothing when unset (a single integer compare gates the whole path) and works under streaming. Code blocks (`<pre>`/`<code>`), tables, and headings are never wrapped; blockquote and list continuations keep their `> ` prefix / marker-width indent. Words are never split, so an oversized token (e.g. a URL) overflows rather than breaking.

Wired through every layer: the Rust core (`HTMLToMarkdownOptions.wrap_width`), the pure-JS engine, the NAPI bindings, the `mdream` wrapper, and both CLIs (`--wrap-width <n>`).

Behavior was fact-checked against Python `html2text` (`body_width`), the de-facto reference for width wrapping. We match it on word boundaries, no long-word splitting, skipping code/tables, and blockquote/list continuation indents. Intentional divergences: wrapping is opt-in (no default 78) to keep the hot path free; headings are not wrapped (html2text can split them); list items are wrapped with correct CommonMark continuation indent (html2text skips them by default).

Tested: Rust core suite (incl. 8 new wrap tests), `packages/js` suite (incl. 7 new wrap tests covering streaming), full `mdream` package suite, and an end-to-end check through the rebuilt native module. All green; typechecks clean.

## Summary by CodeRabbit

* **New Features**
  * Added configurable hard-wrapping of prose during conversion via a new wrapWidth option (exposed as CLI flag --wrap-width and in JS/Node/WASM options). Wrapping breaks on word boundaries, is disabled by default and when set to 0, skips code blocks/tables/headings, and preserves correct list/blockquote continuation prefixes. Streaming conversion honors the setting.

* **Tests**
  * Added comprehensive tests covering wrap behavior, edge cases, nested structures, and streaming parity.

### CODE REVIEW

File: crates/core/src/types.rs [hunk 1]
```diff
@@ -507,6 +507,12 @@ pub struct HTMLToMarkdownOptions {
     pub clean: Option<CleanConfig>,
     /// Plugin configuration (filters, frontmatter, extraction, tag overrides, …).
     pub plugins: Option<PluginConfig>,
+    /// Hard-wrap prose at this many characters, breaking on word boundaries.
+    ///
+    /// Applied inline during conversion (no extra pass), so it is zero-cost when
+    /// unset. `Some(0)` and `None` both disable wrapping. Code (`<pre>`/`<code>`),
+    /// tables, and headings are never wrapped.
+    pub wrap_width: Option<usize>,
```
Comment (xitep) [inline crates/core/src/types.rs:515]: just a minor; i wonder why the `Option` is necessary in the first place. a `0` as a default would fully serve the current semantics i believe. (in fact, a `0` is what we'd get relying on the derived `Default` impl.) maybe this could be simplified as a detail in some of the future releases?
